ci/cron/src/Main.hs

```diff
@@ -308,9 +311,32 @@ instance JSON.FromJSON GitHubVersion where
 name :: GitHubVersion -> String
 name gh = tail $ tag_name gh
 
+fetch_gh_paginated :: JSON.FromJSON a => String -> IO [a]
+fetch_gh_paginated url = do
+    (resp_0, headers) <- http_get url
+    case parse_next $ Map.lookup "link" headers of
+      Nothing -> return resp_0
+      Just next -> do
+          rest <- fetch_gh_paginated next
+          return $ resp_0 ++ rest
+    where parse_next Nothing = Nothing
+          parse_next (Just header) = take_next $ concatMap parse_link $ split header
+          split h = Split.splitOn ", " h
+          link_regex = "<(.*)>; rel=\"(.*)\"" :: String
+          parse_link l =
+              let typed_regex :: (String, String, String, [String])
+                  typed_regex = l Regex.=~ link_regex
+              in
+              case typed_regex of
+                (_, _, _, [url, rel]) -> [(rel, url)]
+                _ -> []
```

Comment:
When do we expect to fall into this case? If the answer is “never”, maybe calling “error” would be better so we know if this fails?

Reply:
I'm actually not sure; the RFC allows for more attributes than `rel`, but I haven't seen any other in GH examples. They do not document their use of the format, that I could find. I guess you're right on the fact that it's better if we get told when that assumption gets invalidated.